### ISSUE TITLE
Profile overlay shows data of previous user when switching users without reload

### ISSUE THREAD

**Issue by jordygroote** [sulu/sulu#5759]
| Q | A
| --- | ---
| Sulu Version | Latest

# Actual Behavior

The edit profile keeps hanging on administrator eventhough i am not logged in as administrator.
Also the form is broken and cannot save.

# Expected Behavior

The edit profile modal is showing the correct info of the account and the form can be saved.

# Steps to Reproduce
- Be logged in as administrator
- Create account
- Login as new account
- Click left top menu bar to open edit profile

I added screenshots (screenshots are from sulu.rocks).
![Schermafbeelding 2021-01-19 om 13 43 39](https://user-images.githubusercontent.com/8653367/105036413-ba0f7b80-5a5c-11eb-8c24-2df6441cae60.png)
![Schermafbeelding 2021-01-19 om 13 43 44](https://user-images.githubusercontent.com/8653367/105036416-bbd93f00-5a5c-11eb-94cf-2b7bddf8b02c.png)

**Comment by alexander-schranz** [comment]
Seems like the form is not cleared. After refresh the new data is shown. I think the overlay should always load the data from the endpoint and not cache it.

**Comment by niklasnatter** [comment]
> Seems like the form is not cleared. After refresh the new data is shown. I think the overlay should always load the data from the endpoint and not cache it.

Sounds good. Solving this might be a good opportunity form implementing a reusable `FormOverlay` container in the core. I implemented something like that in a project recently:

```javascript
/* ResourceFormOverlay.js */

import React from 'react';
import {observer} from 'mobx-react';
import {translate} from 'sulu-admin-bundle/utils/Translator';
import Form from 'sulu-admin-bundle/containers/Form';
import Overlay from 'sulu-admin-bundle/components/Overlay';
import Snackbar from 'sulu-admin-bundle/components/Snackbar';
import ResourceFormStore from 'sulu-admin-bundle/containers/Form/stores/ResourceFormStore';
import ResourceStore from 'sulu-admin-bundle/stores/ResourceStore';
import {action, observable} from 'mobx';
import Loader from 'sulu-admin-bundle/components/Loader';
import resourceFormOverlayStyles from './resourceFormOverlay.scss';

@observer
class ResourceFormOverlay extends React.Component {
    static defaultProps = {
        formStoreMetadataOptions: {},
        formStoreOptions: {},
    };

    formRef;

    @observable formStore;
    @observable formErrors = [];

    handleOverlayConfirm = () => {
        if (!this.formRef) {
            throw new Error('The Form ref has not been set! This should not happen and is likely a bug.');
        }

        this.formRef.submit();
    };

    handleFormSubmit = () => {
        const {onSaved} = this.props;

        if (!this.formStore) {
            throw new Error('The FormStore has not been initialized! This should not happen and is likely a bug.');
        }

        this.formStore.save()
            .then(() => {
                onSaved();
            })
            .catch(action(() => {
                this.formErrors.push(translate('sulu_admin.form_save_server_error'));
            }));
    };

    @action handleFormError = () => {
        this.formErrors.push(translate('sulu_admin.form_contains_invalid_values'));
    };

    @action handleErrorSnackbarClose = () => {
        this.formErrors.pop();
    };

    @action componentDidUpdate(prevProps) {
        const {open} = this.props;

        if (prevProps.open === false && open === true) {
            this.initializeFormData();
        }
    }

    @action initializeFormData = () => {
        const {
            locale,
            formKey,
            resourceKey,
            resourceId,
            formStoreOptions,
            formStoreMetadataOptions,
        } = this.props;

        this.destroyFormData();

        const observableOptions = {};
        if (locale) {
            observableOptions.locale = locale;
        }

        const resourceStore = new ResourceStore(resourceKey, resourceId, observableOptions, formStoreOptions);
        this.formStore = new ResourceFormStore(resourceStore, formKey, formStoreOptions, formStoreMetadataOptions);
    };

    @action destroyFormData = () => {
        this.formErrors = [];

        if (this.formStore) {
            this.formStore.destroy();
            this.formStore = undefined;
        }
    };

    setFormRef = (formRef) => {
        this.formRef = formRef;
    };

    componentWillUnmount() {
        this.destroyFormData();
    }

    render() {
        const {saveText, open, size, title, onClose} = this.props;

        return (
            <Overlay
                confirmDisabled={!this.formStore || !this.formStore.dirty}
                confirmLoading={this.formStore && this.formStore.saving}
                confirmText={saveText}
                onClose={onClose}
                onConfirm={this.handleOverlayConfirm}
                open={open}
                size={size}
                title={title}
            >
                <Snackbar
                    message={this.formErrors[this.formErrors.length - 1]}
                    onCloseClick={this.handleErrorSnackbarClose}
                    type="error"
                    visible={!!this.formErrors.length}
                />
                <div className={resourceFormOverlayStyles.overlay}>
                    {!this.formStore && (
                        <Loader />
                    )}
                    {!!this.formStore && (
                        <Form
                            onError={this.handleFormError}
                            onSubmit={this.handleFormSubmit}
                            ref={this.setFormRef}
                            store={this.formStore}
                        />
                    )}
                </div>
            </Overlay>
        );
    }
}

export default ResourceFormOverlay;
```

```scss
/* resourceFormOverlay.scss */

@import 'sulu-admin-bundle/containers/Application/variables.scss';

.overlay {
    padding: $viewPaddingVertical $viewPaddingHorizontal;;
}
```

**Comment by niklasnatter** [comment]
Thanks for reporting! Fixed by #5906
